src/xss-filters.js

```diff
@@ -251,6 +251,57 @@ exports._getPrivFilters = function () {
         }
         return null;
     }
+
+    /* 
+     * Create URL whitelist filter
+     * Ref: https://url.spec.whatwg.org/#url-parsing
+     * @param {boolean} allowAllSafeURIs - to enable inherited protocol (//), blob:, http://, https://, ftp://, "data:image/[gif,jpeg,png];base64," are allowed
+     * @param {boolean} allowInheritProto - to enable inherited protocol (//), or otherwise, only http:// and https:// are allowed
+     * @param {boolean} allowRelPath - to allow relative path
+     * @param {boolean} allowHosts - an optional array of hostnames that each matches /^[\w\.-]+$/. If any one is found unmatched, return null
+     * @param {boolean} reqHtmlDecode - to run html decoding first before applying the tests
+     * @returns {function|null} the yuwl filter that tests value according to the config
+     */
+    function yuwlFactory (allowAllSafeURIs, allowInheritProto, allowRelPath, allowHosts, reqHtmlDecode) {
+        var i, n, reHost, reHosts,
+            reProto = allowAllSafeURIs ? 
+                        /^[\x00-\x20]*(?:(?:(blob:)?https?|ftp):\/\/|data:image\/(?:gif|jpe?g|png);base64,|\/\/)/i :
```

Comment:
can we also allow configurable set of schemes? users might want to have other common schemes like mailto, xmpp or even their own custom schemes